app/frontendapi.py

```diff
@@ -0,0 +1,112 @@
+from django.http import JsonResponse
+
+import config
+from model import Person
+import photo
+import search
+import simplejson
+import utils
+
+
+class FrontendApiHandler(utils.BaseHandler):
+    """Base handler for frontend API handlers."""
+
+    def __init__(self, request, response, env):
+        utils.BaseHandler.__init__(self, request, response, env)
+
+    def _return_json(self, data):
+        self.response.headers['Content-Type'] = (
+            'application/json; charsetutf-8')
+        self.write(simplejson.dumps(data))
+
+
+class Repo(FrontendApiHandler):
+
+    repo_required = False
+
+    def get(self):
+        # TODO: implement this for real
+        if self.env.repo:
+            json = {
+                'repoId': 'haiti',
+                'title': 'Haiti earthquake',
+                'recordCount': 12345
+            }
+        else:
+            json = [
+                {
+                    'repoId': 'haiti',
+                    'title': 'Haiti earthquake',
+                    'recordCount': 12345,
+                },
+                {
+                    'repoId': 'japan',
+                    'title': 'Japan earthquake',
+                    'recordCount': 54321,
+                },
+            ]
+        self._return_json(json)
+
+
+class Results(FrontendApiHandler):
+
+    repo_required = True
+
+    def __init__(self, request, response, env):
+        FrontendApiHandler.__init__(self, request, response, env)
```

Comment:
Optional: Alternatively this can be written as:
```python
super(Results, self).__init__(self, request, response, env)
```
I guess this is more common and it's clearer that it's calling a superclass?